src/ai/backend/manager/scheduler/predicates.py

```diff
@@ -237,3 +243,78 @@ async def check_domain_resource_limit(
             ),
         )
     return PredicateResult(True)
+
+
+async def check_pending_session_limit(
+    db_sess: SASession,
+    sched_ctx: SchedulingContext,
+    sess_ctx: SessionRow,
+) -> PredicateResult:
+    result = True
+    failure_msgs = []
+
+    kernel_query = (
+        sa.select([kernels.c.occupied_slots])
+        .select_from(kernels)
+        .where(
+            (kernels.c.status == KernelStatus.PENDING)
+            & (kernels.c.cluster_role == DEFAULT_ROLE)
+            & (kernels.c.access_key == sess_ctx.access_key)
+        )
+    )
+    pending_kernels: list[dict[str, Any]] = (await db_sess.execute(kernel_query)).fetchall()
```

Comment:
Doesn't the session rows have the pre-calculated sum of the resource slots of all its own kernels??